### PR TITLE
Workload ID: Add Protos for WorkloadIdentity resource, CRUD and events

### PR DESCRIPTION
**Pull request by strideynet**
Adds protos for the new WorkloadIdentity resource, along with the Audit Event protos and similar.

I've gone with the most basic version of the protos presented in the RFD for now. I figure it makes sense to move forward with this boilerplatery work whilst the finer points of the RFD are discussed.

As per RFD: https://github.com/gravitational/teleport/pull/49133

Part of https://github.com/gravitational/teleport/issues/49572

### CODE REVIEW

File: api/proto/teleport/legacy/types/events/events.proto [hunk 1]
```diff
@@ -7677,3 +7680,96 @@ enum ContactType {
...
+    (gogoproto.embed) = true,
+    (gogoproto.jsontag) = ""
+  ];
+
+  // ConnectionMetadata holds information about the connection
+  ConnectionMetadata Connection = 4 [
+    (gogoproto.nullable) = false,
+    (gogoproto.embed) = true,
+    (gogoproto.jsontag) = ""
+  ];
```
Comment (tigrato) [inline api/proto/teleport/legacy/types/events/events.proto:7712]: Add the resource payload

File: api/proto/teleport/workloadidentity/v1/resource_service.proto [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import "teleport/workloadidentity/v1/resource.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/workloadidentity/v1;workloadidentityv1pb";
+
+// WorkloadIdentityResourceService provides methods to manage the
+// WorkloadIdentity resource.
+service WorkloadIdentityResourceService {
+  // CreateWorkloadIdentity creates a new workload identity, it will refuse to
+  // create a workload identity if one already exists with the same name.
+  rpc CreateWorkloadIdentity(CreateWorkloadIdentityRequest) returns (teleport.workloadidentity.v1.WorkloadIdentity);
```
Comment (tigrato) [inline api/proto/teleport/workloadidentity/v1/resource_service.proto:29]:
```suggestion
  rpc CreateWorkloadIdentity(CreateWorkloadIdentityRequest) returns (WorkloadIdentity);
```

Should we hide the package given it's the same package?

File: api/proto/teleport/workloadidentity/v1/resource_service.proto [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  // UpdateWorkloadIdentity updates an existing workload identity, it will refuse
+  // to update a workload identity if one does not already exist with the same name.
+  rpc UpdateWorkloadIdentity(UpdateWorkloadIdentityRequest) returns (teleport.workloadidentity.v1.WorkloadIdentity);
```
Comment (tigrato) [inline api/proto/teleport/workloadidentity/v1/resource_service.proto:32]: Should we include the upsert?

File: api/proto/teleport/legacy/types/events/events.proto (hunk 1 above)
Comment (strideynet) [reply api/proto/teleport/legacy/types/events/events.proto:7712]: [09801ec](https://github.com/gravitational/teleport/pull/49650/commits/09801ec8512b1e56781e8fe4a646c5504ca259b0)

File: api/proto/teleport/workloadidentity/v1/resource_service.proto (hunk 2 above)
Comment (strideynet) [reply api/proto/teleport/workloadidentity/v1/resource_service.proto:29]: 9bc819a4ce

**Review by tigrato — APPROVED** [review]

File: api/types/events/events.go [hunk 4]
```diff
@@ -2360,3 +2360,15 @@ func (m *ContactCreate) TrimToMaxSize(_ int) AuditEvent {
 func (m *ContactDelete) TrimToMaxSize(_ int) AuditEvent {
 	return m
 }
+
+func (m *WorkloadIdentityCreate) TrimToMaxSize(_ int) AuditEvent {
```
[the remark this answers is not in the document]
Comment (strideynet) [reply api/types/events/events.go]: 36bcd96a68

**Review by timothyb89 — APPROVED** [review]

**Review by capnspacehook — APPROVED** [review]
Thanks for adding the `TrimToMaxSize` impl!

**Pull request merged**